test/test_cuda.py

```diff
@@ -4659,9 +4441,212 @@ def test_no_triton_on_import(self):
         self.assertEqual(rc, "False", "Triton was imported when importing torch!")
 
 
+@torch.testing._internal.common_utils.markDynamoStrictTest
 class TestCudaOptims(TestCase):
     # These tests will be instantiate with instantiate_device_type_tests
     # to apply the new OptimizerInfo structure.
+    def _test_graphed_optims(self, steps_warmup, steps_train, optimizer_ctor, kwargs):
+        for actually_do_graphs in (True, False):
+            params = [torch.randn((i + 5, i + 5), device="cuda") for i in range(2)] + [
+                torch.randn((), device="cuda")
+            ]
+            params_control = [p.clone().requires_grad_() for p in params]
+            params_graphed = [p.clone().requires_grad_() for p in params]
+
+            grads = [
+                [torch.randn_like(p) for p in params]
+                for _ in range(steps_warmup + steps_train)
+            ]
+
+            # Control (capturable=False)
+            if "capturable" in kwargs:
+                del kwargs["capturable"]
```

Comment:
```suggestion
```
For this test, it'd make sense to only test for optim in optim_db if optim.has_capturable_arg

Reply:
And here, I'd do `kwargs["capturable"] = True`
so you don't modify line 4464